akd_local_auditor/src/storage/s3/test.rs

```diff
@@ -0,0 +1,368 @@
+// and the s3.yml workflow pipeline
+const ACCESS_KEY: &str = "test";
+const SECRET_KEY: &str = "someLongAccessKey";
+const TEST_REGION: &str = "us-east-2";
+const TEST_ENDPOINT: &str = "http://127.0.0.1:9000";
+
+macro_rules! bucket_name {
+    () => {{
+        fn f() {}
+        fn type_name_of<T>(_: T) -> &'static str {
+            std::any::type_name::<T>()
+        }
+        let name = type_name_of(f);
+        name[..name.len() - 3]
+            .trim_end_matches("::{{closure}}")
+            .split("::")
+            .last()
+            .unwrap()
+            .replace("_", "")
+    }};
+}
+
+struct AuditInformation {
+    proof: akd::proof_structs::AppendOnlyProof<Hasher>,
+    phash: Digest,
+    chash: Digest,
+}
+
+/// Generate N audit proofs from a new tree
+async fn generate_audit_proofs(n: usize) -> Result<Vec<AuditInformation>, akd::errors::AkdError> {
+    let db = AsyncInMemoryDatabase::new();
+    let vrf = HardCodedAkdVRF {};
+    let akd = Directory::<_, _>::new::<Hasher>(&db, &vrf, false).await?;
+    let mut proofs = vec![];
+    // gather the hash + azks for epoch "0" (init)
+    let mut azks = akd.retrieve_current_azks().await?;
+    let mut phash = akd.get_root_hash::<Hasher>(&azks).await?;
+
+    for _epoch in 1..=n {
+        // generate (n) epochs of updates on the same user
+        let t_value = format!("certificate{}", _epoch);
+        akd.publish::<Hasher>(vec![(
+            AkdLabel::from_utf8_str("user"),
+            AkdValue::from_utf8_str(&t_value),
+        )])
+        .await?;
+        // generate the append-only proof
+        let proof = akd
+            .audit::<Hasher>((_epoch - 1) as u64, _epoch as u64)
+            .await?;
+        // update the azks + retrieve the "next" hash
+        azks = akd.retrieve_current_azks().await?;
+        let chash = akd.get_root_hash::<Hasher>(&azks).await?;
+
+        // we have everything we need, pass it along
+        let info = AuditInformation {
+            phash,
+            chash,
+            proof,
+        };
+        proofs.push(info);
+
+        // reset the "previous" hash for the next iteration
+        phash = chash;
+    }
+    Ok(proofs)
+}
+
+/// Flush the storage bucket of all blobs + delete the bucket (if it exists)
+async fn maybe_flush_storage(shared_config: &SdkConfig, bucket_name: &str) -> Result<()> {
+    // Build the S3 config from the shared SdkConfig
+    let config = S3ConfigBuilder::from(shared_config)
+        .retry_config(RetryConfig::disabled())
+        .build();
+    // get the S3 client
+    let client = Client::from_conf(config);
+    // check if the bucket exists
+    let buckets = client.list_buckets().send().await?;
+
+    if let Some(bucket_names) = buckets.buckets() {
+        log::info!("Found {} buckets in the test storage", bucket_names.len());
+
+        for bucket in bucket_names {
+            if let Some(maybe_name) = bucket.name() {
+                if maybe_name == bucket_name {
```

Comment:
Seeing as how we're iterating over buckets to find the first one here, I think you can use [`find`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find) on the iterator to get an option of the first result. From there, you can map it to do what you're doing in this loop. I _think_ that should be slightly cleaner, but ultimately you're call!